sympy/physics/control/control_plots.py

```diff
@@ -23,12 +22,23 @@
 
 numpy = import_module('numpy')
 
+sympy_plot_backends = import_module('sympy_plot_backends')
+
 if matplotlib:
     plt = matplotlib.pyplot
 
 if numpy:
     np = numpy  # Matplotlib already has numpy as a compulsory dependency. No need to install it separately.
 
+if sympy_plot_backends:
+    spb = sympy_plot_backends
+
+# Imports from sympy-plot-backends.
+# TODO : Fix Imports after new plotting module is merged.
+from spb import plot_list
+from spb.backends.matplotlib.renderers.line2d import (
+   Line2DRenderer, _draw_line2d_helper, _update_line2d_helper
+)
```

Comment:
The issue to the failing tests lies here. I want to use functions from the sympy-plot-backends library.
But I am not aware how to ensure that it gets installed for the user before (maybe some requirements file in plotting ?). I did `pip install sympy_plot_backends` locally so it's runs well and no tests fail.

Hi @sylee957, could you please have a look and help me here . Thanks in advance !

Reply:
Now that I fully understand [your question](https://github.com/Davide-sd/sympy-plot-backends/issues/28), the approach I suggested yesterday is far from optimal because it would also affect all other plotting functions that uses `Line2DRenderer`, which is inconvenient. 

A better way to achieve that is by creating new data series for the axes, and add them to the existing plot, like this:

```py
from spb.series import HVLineSeries
x = [1, -2, 3]
y = [2, 5, -6]
p = plot_list(x, y, is_point=True, show=False, grid=False)
hor = HVLineSeries(0, True, show_in_legend=False)
ver = HVLineSeries(0, False, show_in_legend=False)
p.extend([hor, ver])
p.show()
```

In doing so, the following code can be removed:

```py
from spb.backends.matplotlib.renderers.line2d import (
   Line2DRenderer, _draw_line2d_helper, _update_line2d_helper
)
```

as well as the code that uses these functionalities.

Reply:
Now, let's address the elephant in the room. Before your GSoC, the plotting function from the control module created a Matplotlib figure and returned Matplotlib's pyplot module (`plt` in the code). It seems like you are changing them to return plot objects created with a plotting module.

Here is the problem: if I am a SymPy user, I expect these function to returns plot objects created with SymPy's plotting module. Instead, `zero_pole_plot` will return a plot object created with the [Sympy Plotting Backends](https://github.com/Davide-sd/sympy-plot-backends) module, which is not compatible with plot objects created with the `sympy.plotting` module. Hence, if a user attempts to combine two or more plots created with these two module, an error will be raised.

If you really want to return plotting objects, you should use SymPy's plotting module, which means to modify `List2DSeries` and `MatplotlibBackend` to create a scatter plot. Alternatively, make it very clear in the documentation that you are using an external plotting module which is not fully compatible with SymPy's one.

Reply:
Thanks for all assistance. Having primarily used `Matplotlib` for plotting till now, it will take some time to get familiar with `sympy-plot-backends` but I'm sure I'll get there 👍

Reply:
@faze-geek Note that I released a new version of `spb` to address a bug. You should download it.

But let me be extremely clear: I don't think you should use `spb` here! Instead, you should modify `sympy.plotting.MatplotlibBackend` and `sympy.plotting.List2DSeries` to achieve your goal.

Reply:
>Instead, you should modify `sympy.plotting.MatplotlibBackend` and `sympy.plotting.List2DSeries` to achieve your goal.

Okay, I'll look into this soon.